### PR TITLE
Enable ContentFormatter formatting struct and array type properties with 'undefined' value

### PR DESCRIPTION
**Pull request by JonasDov**

### CODE REVIEW

File: presentation/common/src/presentation-common/content/PropertyValueFormatter.ts [hunk 1]
```diff
@@ -73,11 +73,11 @@ export class ContentFormatter {
   }
 
   private async formatPropertyValue(value: Value, field: PropertiesField): Promise<DisplayValue> {
-    if (field.isArrayPropertiesField()) {
+    if (field.isArrayPropertiesField() && value !== undefined) {
```
Comment (saskliutas) [inline presentation/common/src/presentation-common/content/PropertyValueFormatter.ts:76]: I think `value !== undefined` check can be moved to `ContentFormatter.formatValues` function to make sure that we do not try formatting value if it is `undefined`.

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: presentation/common/src/presentation-common/content/PropertyValueFormatter.ts (hunk 1 above)
Comment (JonasDov) [reply presentation/common/src/presentation-common/content/PropertyValueFormatter.ts]: Moved

File: presentation/common/src/presentation-common/content/PropertyValueFormatter.ts [hunk 2]
```diff
@@ -49,6 +49,11 @@ export class ContentFormatter {
         continue;
       }
 
+      // do not add undefined value to display values, it's raw value will be formatted
```
Comment (saskliutas) [inline presentation/common/src/presentation-common/content/PropertyValueFormatter.ts:52]:
```suggestion
      // do not add undefined value to display values
```

**Review by saskliutas — APPROVED** [review]

**Comment by saskliutas** [comment]
@mergifyio backport release/4.9.x

**Pull request merged**